### PR TITLE
Replace 0L with an UNASSIGNED_PRIMARY_TERM constant

### PR DESCRIPTION
**Pull request by bleskes**
0 is an illegal value for a primary term that is often used to indicate
the primary term isn't assigned or is irrelevant. This PR replaces the
usage of 0 with a constant, to improve readability and so it can be
tracked and if needed, replaced.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/index/IndexRequest.java [hunk 1]
```diff
@@ -174,7 +175,7 @@ public ActionRequestValidationException validate() {
                 return validationException;
             }
 
-            if (ifSeqNo != SequenceNumbers.UNASSIGNED_SEQ_NO || ifPrimaryTerm != 0) {
+            if (ifSeqNo != UNASSIGNED_SEQ_NO || ifPrimaryTerm != 0) {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: there's another 0 here

File: server/src/main/java/org/elasticsearch/action/index/IndexRequest.java [hunk 2]
```diff
@@ -657,7 +658,7 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_6_6_0)) {
             out.writeZLong(ifSeqNo);
             out.writeVLong(ifPrimaryTerm);
-        } else if (ifSeqNo != SequenceNumbers.UNASSIGNED_SEQ_NO || ifPrimaryTerm != 0) {
+        } else if (ifSeqNo != UNASSIGNED_SEQ_NO || ifPrimaryTerm != 0) {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: another 0 here

File: server/src/main/java/org/elasticsearch/index/seqno/SequenceNumbers.java [hunk 3]
```diff
@@ -41,6 +41,11 @@
      */
     public static final long NO_OPS_PERFORMED = -1L;
 
+    /**
+     * Represents an unassigned primary term (e.g., when a primary shard was not yet allocated)
+     */
+    public static final long UNASSINGED_PRIMARY_TERM = 0L;
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/index/seqno/SequenceNumbers.java]: you had one job 😱

**Review by ywelsch — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/elasticsearch/index/seqno/SequenceNumbers.java (hunk 3 above)
Comment (bleskes) [reply server/src/main/java/org/elasticsearch/index/seqno/SequenceNumbers.java]: LOL....

File: server/src/main/java/org/elasticsearch/action/index/IndexRequest.java (hunk 1 above)
Comment (bleskes) [reply server/src/main/java/org/elasticsearch/action/index/IndexRequest.java]: I missed that pattern. Getting one more regex out of my toolbox.

**Review by ywelsch — APPROVED** [review]
LGTM

**Pull request merged**